app/controllers/mixins/ems_common_angular.rb

```diff
@@ -149,6 +149,8 @@ def get_task_args(ems)
         [user, params[:default_password], endpoint_opts]
       when 'ManageIQ::Providers::Lenovo::PhysicalInfraManager'
         [user, password, params[:default_hostname], params[:default_api_port], "token", false]
+      when 'ManageIQ::Providers::Hawkular::MiddlewareManager'
+        [params[:default_hostname], params[:default_api_port], user, params[:default_password], params[:default_security_protocol], params[:default_tls_ca_certs], true]
```

Comment:
Probably missing call to `ems.ssl_cert_store`,
e.g. 
`ems.ssl_cert_store(params[:default_tls_ca_certs])`

Reply:
No, we can't do this, I am looking for another way or maybe make the conversion in the provider.

Reply:
Maybe with something like
```ruby
    if params[:default_tls_ca_certs].present?
            pem_fragment = params[:default_tls_ca_certs].split(/(?=-----BEGIN)/).reject(&:blank?).collect.first
            cert = OpenSSL::X509::Certificate.new(pem_fragment)
            store = OpenSSL::X509::Store.new
            store.add_cert(cert)
          end
        [params[:default_hostname], params[:default_api_port], user, params[:default_password], params[:default_security_protocol], store, true]
    ```
But I think we should make this in the provider like https://github.com/ManageIQ/manageiq-providers-hawkular/blob/master/app/models/manageiq/providers/hawkular/middleware_manager.rb#L78 .Others providers like RHEV do the conversion in

Reply:
I think it should be done on the provider (make sense to me). I'd also like to see it as a separate PR since it's not caused by this code but rather a regression caused by the change in view (switching from EMS instance call to a class method call).

Reply:
I think so. Do you agree with this @abonas ?

Reply:
I'm not sure how is handled now by other providers, is OK if this can be done in the provider code, the endpoint [1] model is the one that has the specific code.

Can we use that code in the provider? It also considers Settings and the system defaults.

[1] https://github.com/cben/manageiq/blob/1d20c07858d54bb71e640fc96d67c09d11d631b4/app/models/endpoint.rb#L22

Reply:
But endpoint is not used until de provider is created I think and this is the problem, I can't call this functionallity in this step because ems.default_endpoint.certificate_authorities is not defined